Design.md

```diff
@@ -48,10 +48,10 @@ Proposed key generation scheme:
 
 The keys actually used to sotre fragments on the hyperdrives can be derived easily with only the generated key, even for splits.
 ```
-<stored_fragment_key> := <serviceId>-<ctime>-<hash>-<start_offset>-<fragid>
+<stored_fragment_key> := <serviceId>-<ctime>-<hash>-<end_offset>-<fragid>
 <serviceId>, <ctime> and <hash> are the ones defined above
 <fragid>:= index in main key fragment list
-<start_offset> := used for splits. All split chunks share the same prefix, storing the offset is used to easily have range queries and avoid storing them all in the main key.
+<end_offset> := used for splits. All split chunks share the same prefix, storing the offset is used to easily have range queries and avoid storing them all in the main key. End of chunk offset is used to be able to infer object real size from last chunk keys alone (only way to do it for erasure coded last chunk parts).
```

Comment:
Can we be more explicit that the end offset is EXCLUSIVE ? (ie. offset of byte just after the ending) - or did I misunderstand it ?

Reply:
It is indeed exclusive. WIll add a comment